src/App.tsx

```diff
@@ -1,24 +1,156 @@
+import { TodoList } from './components/TodoList';
+import { TodoFilter } from './types/TodoFelter';
+import { ErrorNotification } from './components/ErrorNotification';
+import { Error } from './types/Error';
 
-const USER_ID = 0;
+const USER_ID = 11960;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [currentFilter, setCurrentFilter]
+    = useState<TodoFilter>(TodoFilter.All);
+  const [errorText, setErrorText] = useState<Error>(Error.Default);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => setErrorText(Error.CantLoad));
+  }, []);
+
+  const filteredTodos = useMemo(() => {
+    switch (currentFilter) {
+      case TodoFilter.Active:
+        return todos.filter(todo => !todo.completed);
+      case TodoFilter.Completed:
+        return todos.filter(todo => todo.completed);
+      default:
+        return todos;
+    }
+  }, [todos, currentFilter]);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const handleNewTodo = (todo: Todo) => {
+    setTodos(current => [...current, todo]);
+  };
+
+  const handleDeleteTodo = (todoId: number) => {
+    deleteTodo(todoId)
+      .then(() => {
+        setTodos(current => current.filter(todo => todo.id !== todoId));
+      })
+      .catch(() => setErrorText(Error.Delete));
+  };
+
+  const toggleAllStatus = () => {
+    let status = true;
+
+    if (todos.filter(item => item.completed).length < todos.length) {
+      status = false;
+    }
+
+    return status;
+  };
+
+  const handleDeleteCompleted = async (todosId: number[]) => {
+    try {
+      for (let i = 0; i < todosId.length; i += 1) {
+        deleteTodo(todosId[i]);
+      }
+    } catch (error) {
+      setErrorText(Error.Delete);
+    } finally {
+      setTodos(current => current.filter(todo => !todosId.includes(todo.id)));
+    }
+  };
+
+  const handleUpdateAll = () => {
+    const status = !toggleAllStatus();
+
+    const currentTodos = [...todos];
+
+    for (let i = 0; i < currentTodos.length; i += 1) {
+      if (currentTodos[i].completed !== status) {
+        currentTodos[i].completed = status;
+
+        updateTodo(currentTodos[i])
+          .then(() => {
+            setTodos((current) => {
+              const newTodos = [...current];
+              const index = newTodos.findIndex(
+                (item) => item.id === currentTodos[i].id,
+              );
+
+              newTodos.splice(index, 1, currentTodos[i]);
+
+              return newTodos;
+            });
+          })
+          .catch(() => {
+            setErrorText(Error.Update);
+          });
+      }
+    }
+  };
+
+  const handleUpdateTodo = (todo: Todo) => {
+    setTodos(currentTodos => {
+      const newTodos = [...currentTodos];
+      const index = newTodos.findIndex(item => item.id === todo.id);
+
+      newTodos.splice(index, 1, todo);
+
+      return newTodos;
+    });
```

Comment:
```suggestion
    setTodos(currentTodos => currentTodos.map(currentTodo => {
       if (currentTodo.id === todo.id) return todo;
      return currentTodo;
    }));
```

Reply:
Still not fixed from the previous review